runtime/src/accounts_index_storage.rs

```diff
@@ -57,32 +58,38 @@ impl BgThreads {
         in_mem: &[Arc<InMemAccountsIndex<T>>],
         threads: usize,
         can_advance_age: bool,
+        exit: &Arc<AtomicBool>,
     ) -> Self {
         // stop signal used for THIS batch of bg threads
-        let exit = Arc::new(AtomicBool::default());
+        let local_exit = Arc::new(AtomicBool::default());
         let handles = Some(
             (0..threads)
                 .into_iter()
                 .map(|idx| {
                     // the first thread we start is special
                     let can_advance_age = can_advance_age && idx == 0;
                     let storage_ = Arc::clone(storage);
-                    let exit_ = Arc::clone(&exit);
+                    let local_exit_ = Arc::clone(&local_exit);
+                    let system_exit_ = Arc::clone(exit);
                     let in_mem_ = in_mem.to_vec();
 
                     // note that using rayon here causes us to exhaust # rayon threads and many tests running in parallel deadlock
                     Builder::new()
                         .name(format!("solIdxFlusher{:02}", idx))
                         .spawn(move || {
-                            storage_.background(exit_, in_mem_, can_advance_age);
+                            storage_.background(
+                                vec![local_exit_, system_exit_],
```

Comment:
nit: i'd have expected the exit vec to be constructed above, alongside the other variables that the closure is going to move